### ISSUE TITLE
Custom filter (lua function) for Aura trigger

### ISSUE THREAD

**Issue by Weeeqt** [WeakAuras/WeakAuras2#2285]
**Is your feature request related to a problem? Please describe.**
Sometimes I want to apply a custom filter in the aura trigger, like check if the aura is crowd control by checking if it's spellID is in DRData library, or check if aura passes a set of my own conditions, for example if CompactUnitFrame_UtilShouldDisplayBuff(...) function returns true for this aura.
Right now to achieve this I have to create a custom event trigger, but it would be really nice to have an additional filter in Aura trigger.

**Describe the solution you'd like**
In Aura trigger I'd like to see an additional filter "Custom Filter" or "Custom Function", which, once enabled, will allow us to write a custom function like:
`function(spellId, ...)
   return spellId == 123;
end`
Basically it should run this function and pass aura information as argument(s). If the function return true - trigger will proceed with other conditions/filters. If returns false - it won't trigger.
Also I think this filter should be checked last, after all standard ones.

**Comment by InfusOnWoW** [comment]
That's mostly a duplicate of https://github.com/WeakAuras/WeakAuras2/issues/937, feel free to add to that ticket.
